### ISSUE TITLE
FTBFS docs: sphinx 5.1.0 "pop from an empty deque"

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#6882]
Started to happen only recently and spotted in #6879 first:

```shell
$> git grep 'threw an exception (exception: pop from an empty deque)'
24/pr/6879/3a40934/github-Docs-5343-failed/1_build.txt:2022-07-24T14:33:00.9989707Z Handler <function _process_docstring at 0x7fc4eefe9170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
24/pr/6879/3a40934/github-Docs-5343-failed/build/6_Build docs.txt:2022-07-24T14:33:00.9989681Z Handler <function _process_docstring at 0x7fc4eefe9170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
24/pr/6880/ca7460f/github-Docs-5344-failed/1_build.txt:2022-07-24T17:02:14.7290695Z Handler <function _process_docstring at 0x7f37f80ea170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
24/pr/6880/ca7460f/github-Docs-5344-failed/build/6_Build docs.txt:2022-07-24T17:02:14.7290665Z Handler <function _process_docstring at 0x7f37f80ea170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
24/pr/6881/5ae1b46/github-Docs-5345-failed/1_build.txt:2022-07-24T17:18:28.8465361Z Handler <function _process_docstring at 0x7f88229f5170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
24/pr/6881/5ae1b46/github-Docs-5345-failed/build/6_Build docs.txt:2022-07-24T17:18:28.8465331Z Handler <function _process_docstring at 0x7f88229f5170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)

```

https://github.com/datalad/datalad/runs/7488517165?check_suite_focus=true :+1: 

```
reading sources... [ 54%] generated/datalad.customremotes.archives
Extension error (sphinx.ext.napoleon):
Handler <function _process_docstring at 0x7fc4eefe9170> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
make: *** [Makefile:50: html] Error 2
make: Leaving directory '/home/runner/work/datalad/datalad/docs'
```

seems due to sphinx upgrade to 5.1.0

```
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/07[master]24
$> git grep 'installed.*sphinx-' | sed -e 's,.*installed.* sphinx-\([0-9.]*\) .*,\1,g' | sort | uniq -c
     70 5.0.2
     53 5.1.0
1 6658.....................................:Sun 24 Jul 2022 01:26:41 PM EDT:.
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/07[master]24
$> cd ../23
pr/
1 6659.....................................:Sun 24 Jul 2022 01:26:45 PM EDT:.
(git)smaug:/mnt/datasets/datalad/ci/logs/2022/07[master]23
$> git grep 'installed.*sphinx-' | sed -e 's,.*installed.* sphinx-\([0-9.]*\) .*,\1,g' | sort | uniq -c
    135 5.0.2
```

**Comment by yarikoptic** [comment]
trigger -- `@property` without docstring. Followed up upstream: https://github.com/sphinx-doc/sphinx/issues/10701#issuecomment-1193365979

**Comment by adswa** [comment]
addressed in  #6883
